ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -265,26 +262,14 @@ int mca_pml_ucx_open(void)
     }
 
     ompi_pml_ucx.request_size     = attr.request_size;
-    ompi_pml_ucx.cuda_initialized = false;
 
-#if HAVE_UCP_ATTR_MEMORY_TYPES && OPAL_CUDA_SUPPORT
-    if (attr.memory_types & UCS_BIT(UCS_MEMORY_TYPE_CUDA)) {
-        mca_common_cuda_stage_one_init();
```

Comment:
Yes, it has dl_open logic during the component init phase, see https://github.com/open-mpi/ompi/pull/10069/files#diff-fb4497b2fa6fedf5c529388bfd346a5561f3c6555319f14f3d53aca44ab31f0fR298

It doesn't need to be called multiple times because the init is just called during opal init. I haven't tested with PML UCX, but iirc PML UCX didn't actually need common_cuda, but maybe someone who worked on pml ucx could elaborate on that better.

Reply:
Yes for testing, Akshay helped ttest that path. Keeping with dl_open right now but looking into linking libcuda.so.1 directly.